src/service/dashboards/mod.rs

```diff
@@ -149,34 +149,26 @@ pub async fn list_dashboards(
 }
 
 #[tracing::instrument]
-pub async fn get_dashboard(
-    org_id: &str,
-    dashboard_id: &str,
-    folder_id: &str,
-) -> Result<Dashboard, DashboardError> {
-    table::dashboards::get(org_id, folder_id, dashboard_id)
+pub async fn get_dashboard(org_id: &str, dashboard_id: &str) -> Result<Dashboard, DashboardError> {
+    table::dashboards::get(org_id, dashboard_id)
         .await?
         .ok_or(DashboardError::DashboardNotFound)
+        .map(|(_f, d)| d)
 }
 
 #[tracing::instrument]
-pub async fn delete_dashboard(
-    org_id: &str,
-    dashboard_id: &str,
-    folder_id: &str,
-) -> Result<(), DashboardError> {
-    match table::dashboards::get(org_id, folder_id, dashboard_id).await? {
-        Some(_) => {} // Dashboard exists. Continue with deleting.
-        None => return Err(DashboardError::DashboardNotFound),
+pub async fn delete_dashboard(org_id: &str, dashboard_id: &str) -> Result<(), DashboardError> {
+    let Some((folder, _dash)) = table::dashboards::get(org_id, dashboard_id).await? else {
+        return Err(DashboardError::DashboardNotFound);
     };
-    table::dashboards::delete(org_id, folder_id, dashboard_id).await?;
+    table::dashboards::delete(org_id, dashboard_id).await?;
     remove_ownership(
         org_id,
         "dashboards",
         Authz {
             obj_id: dashboard_id.to_owned(),
             parent_type: "folders".to_owned(),
-            parent: folder_id.to_owned(),
+            parent: folder.folder_id.to_owned(),
```

Comment:
Use the folder ID from the just-deleted folder rather than from the function parameter